frontstage/views/account/account_delete.py

```diff
@@ -0,0 +1,23 @@
+import logging
+
+from flask import render_template, request, url_for
+from structlog import wrap_logger
+from werkzeug.utils import redirect
+
+from frontstage.common.authorisation import jwt_authorization
+from frontstage.controllers import party_controller
+from frontstage.controllers.auth_controller import delete_account
+from frontstage.views.account import account_bp
+
+logger = wrap_logger(logging.getLogger(__name__))
+
+
+@account_bp.route("/delete", methods=["GET", "POST"])
+@jwt_authorization(request)
+def delete_user_account(session):
+    party_id = session.get_party_id()
+    respondent_details = party_controller.get_respondent_party_by_id(party_id)
```

Comment:
Thinking about it.  There should be a final check somewhere in this function to make sure they really don't have any surveys attached to them.
Nothing has been done to stop me knowing this delete endpoint exists, going to it manually via url and pressing the delete button.

Reply:
👍